R/zchunk_LA116.geo.R

```diff
@@ -29,57 +33,100 @@ module_energy_LA116.geo_DISABLED <- function(command, ...) {
-    # Note that all precursor names (in `add_precursor`) must be in this chunk's inputs
-    # There's also a `same_precursors_as(x)` you can use
-    # If no precursors (very rare) don't call `add_precursor` at all
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+    if( nrow(unique(A16.geo_curves[ c("grade", "extractioncost")])) > length( unique( A16.geo_curves$grade ) ) ){
+      stop( "The geothermal (hydrothermal) supply curves have regionally differentiated price points" )
+    }
+    if( nrow(unique(A16.EGS_curves[ c("grade", "extractioncost")])) > length( unique( A16.EGS_curves$grade ) ) ){
+      stop( "The geothermal (EGS) supply curves have regionally differentiated price points" )
+    }
+
+    # Downscale GCAM 3.0 geothermal resources to countries on the basis of land area
+    # Calculate land area shares of countries within region_GCAM3
+    # if the last of the HISTORICAL_YEARS is in the land area data set, use that,
+    # otherwise just use the most recent year in the land area data
+    if(max(HISTORICAL_YEARS) %in% Land_type_area_ha$year) {
+      geo_land_year = max(HISTORICAL_YEARS)
+    } else {
+      geo_land_year = max(Land_type_area_ha$year)
+    }
+
+    Land_type_area_ha %>%
+      filter(year == geo_land_year) %>%
+      group_by( iso ) %>%
+      summarise(value = as.numeric(sum(value))) %>%
+      left_join_error_no_match(iso_GCAM_regID[c("iso", "region_GCAM3", GCAM_REGION_ID)], by = "iso") ->
+      L116.land_ctry
+
+    L116.land_ctry %>%
+      group_by(region_GCAM3) %>%
+      summarise(sumvalue = sum(value)) ->
+      L116.land_rg3
+
+    L116.land_ctry %>%
+      full_join(L116.land_rg3, by = "region_GCAM3") %>%
+      mutate(share = value / sumvalue) %>%
+      select( -value, -sumvalue ) ->
+      L116.land_share_ctry_rg3
+
+    # Repeat land area shares by the number of grades in the hydrothermal geothermal resource assumptions
+    # and multiply by the available resource quantities
+    L116.land_share_ctry_rg3 %>%
+      repeat_add_columns(unique(A16.geo_curves["grade"] ) ) %>%
+      left_join_error_no_match(A16.geo_curves, by = c( "region_GCAM3", "grade" ) ) %>%
+      mutate(available = share * available) ->
+      L116.geothermal_ctry
+
+    # Aggregate country-level hydrothermal geothermal resource supply curves by GCAM region
+    L116.geothermal_ctry %>%
+      group_by_(GCAM_REGION_ID, "resource", "subresource", "grade", "extractioncost") %>%
+      summarise(available = sum(available)) ->
+      L116.geothermal_rgn
+
+    # Specify the names of the resource table that will be written out
+    L116.table.names <- c( GCAM_REGION_ID, "resource", "subresource", "grade", "available", "extractioncost" )
+    as_tibble( L116.geothermal_rgn[ L116.table.names ]) %>%
```

Comment:
Why not just `L116_geothermal_rgn %>% select(GCAM_REGION_ID, resource, sub resource, grade, available, extractioncost) %>%` ...seems clearer?